substrate/ed25519/src/lib.rs

```diff
@@ -80,31 +96,32 @@ impl Pair {
 		let pkcs8_bytes = signature::Ed25519KeyPair::generate_pkcs8(&rng).unwrap();
 		Pair(signature::Ed25519KeyPair::from_pkcs8(untrusted::Input::from(&pkcs8_bytes)).unwrap())
 	}
+
 	/// Make a new key pair from a seed phrase.
 	pub fn from_seed(seed: &[u8; 32]) -> Pair {
 		Pair(signature::Ed25519KeyPair::from_seed_unchecked(untrusted::Input::from(&seed[..])).unwrap())
 	}
-	/// Make a new key pair from the raw secret.
-	pub fn from_secret(secret: &[u8; 32]) -> Pair {
-		let mut pkcs8_bytes: Vec<_> = FromHex::from_hex("302e020100300506032b657004220420").unwrap();
-		pkcs8_bytes.extend_from_slice(&secret[..]);
-		Pair(signature::Ed25519KeyPair::from_pkcs8_maybe_unchecked(untrusted::Input::from(&pkcs8_bytes)).unwrap())
-	}
+
+	/// Synonym for `from_seed`. @deprecated
+	pub fn from_secret(seed: &[u8; 32]) -> Pair { Self::from_seed(seed) }
```

Comment:
`#[deprecated]`
https://stackoverflow.com/a/37766834/1745195

but given that it's not actually released we could just remove?
it also does seem useful; what's the rationale for deprecation?